lib/trento/application/usecases/hosts/hosts.ex

```diff
@@ -36,4 +47,41 @@ defmodule Trento.Hosts do
         subscription_count
     end
   end
+
+  @spec deregister_host(Ecto.UUID.t(), DateService) :: :ok | {:error, any}
+  def deregister_host(host_id, date_service \\ DateService) do
+    case Repo.get_by(HostReadModel, id: host_id) do
+      nil ->
+        {:error, :host_not_found}
+
+      _ ->
+        maybe_dispatch_host_deregistration_request(host_id, date_service)
+    end
+  end
+
+  defp commanded,
+    do: Application.fetch_env!(:trento, Trento.Commanded)[:adapter]
+
+  defp maybe_dispatch_host_deregistration_request(host_id, date_service) do
+    now = date_service.utc_now()
+    total_deregistration_debounce = @heartbeat_interval + @deregistration_debounce
+
+    query =
+      from(h in Heartbeat,
+        where:
+          h.timestamp >
+            ^DateTime.add(now, -total_deregistration_debounce, :millisecond) and
```

Comment:
Going back in time with the `-` here hehe
I find this "negative" logic a bit confusing, but well, I guess it works.

I found the `h.timestamp + total_debounce < now` and sending the command in `true` easier.
But not a hard opinion, leave it as you best think